backend/src/services/v2/schema.ts

```diff
@@ -1,7 +1,21 @@
+import { MongoServerError } from 'mongodb'
+
 import { testDeploymentSchema, testModelSchema } from '../../../test/testUtils/testModels.js'
 import Schema, { SchemaInterface } from '../../models/v2/Schema.js'
 import { SchemaKindKeys } from '../../types/v2/enums.js'
-import { NotFound } from '../../utils/v2/error.js'
+import { BadReq, NotFound } from '../../utils/v2/error.js'
+
+function isMongoServerError(err: unknown): err is MongoServerError {
```

Comment:
Nice, but can we move these error guards somewhere else?  This feels like a generic function.